plugins/modules/packaging/language/npm.py

```diff
@@ -263,6 +272,7 @@ def main():
     ignore_scripts = module.params['ignore_scripts']
     unsafe_perm = module.params['unsafe_perm']
     ci = module.params['ci']
+    no_optional = module.params['no_optional']
```

Comment:
You need to pass this to the `Npm` class.

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/1428/commits/7d495beb0c8683191e64c9a6647f03232d07acfd.